### DISCUSSION TITLE
iOS 4.3.43 多段动画衔接播放动画播放不全

### DISCUSSION THREAD

**Discussion by EleanWei** [Tencent/libpag#2430] [Q&A] [answered]
你好，iOS项目中需要将多段动画进行连续衔接播放，通过添加Listener监听onAnimationEnd，上一段动画播放结束后紧接着播放下一段。动画比较少的情况没有问题，但是多了之后出现了最后几段动画高频偶现播放不完整，动画刚出现立马就结束了。
    请问有什么方法可以解决这个问题，希望能尽快收到回复，非常感谢！

播放动画的部分代码如下：
    private func playAnimation(path: String, isLoop: Bool, needObserve: Bool) {
        animationView.remove(self)
        if animationView.isPlaying() {
            animationView.pause()
        }
        animationView.setRepeatCount(isLoop ? 0 : 1)
        animationView.setPath(path)
        if needObserve {
            animationView.add(self)
        }
        animationView.play()
    }
效果可以看视频：第一段背景有花朵出现，和最后一段背景花朵就一瞬间结束了。
https://github.com/user-attachments/assets/d3f6fa89-c082-4f75-938e-f5469e5811d1

**Comment by kevingpqi123** [comment, marked as answer]
此类问题建议先搜索下 discussion
 https://github.com/Tencent/libpag/discussions/2138

**Reply by EleanWei** [reply]
相关帖子我看了，有个问题是业务场景对于动画的个数，动画的播放时长，都是不确定的，这种情况有什么好的办法解决的。

**Reply by EleanWei** [reply]
另外，我将sdk版本升级到最新的， 尝试使用多个animationView切换着播放pag文件，保证每一个animationView都不会出现连续链接播放超过3个文件，实测这个问题依然没有解决。

**Reply by kevingpqi123** [reply]
你还是没有明白上面的帖子，pag 的组合模式是动态拼装的，创建一个 PAGComposition，然后可以动态添加多个 pag 文件，动态设置每个 pag 文件的 startTime，只要规则可以通过代码实现出来就行，然后通过一个 PAGView 播放就可以
